### ISSUE TITLE
Q: Whether forward proxy check url 

### ISSUE THREAD

**Issue by oceanwalker** [apache/trafficserver#8938]
Recently found CVE-2021-44040 vulnerabilities, I wonder if forward proxy check url and involved this vulnerabilities.

I review the commits between 8.1.3 and 8.1.4, I found only this config related to vulnerabilities, so please help me, thks!
.. ts:cv:: CONFIG proxy.config.http.strict_uri_parsing INT 0
.. ts:cv:: CONFIG proxy.config.http.strict_uri_parsing INT 2

**Comment by ezelkow1** [comment]
Yes that change was in relation to that CVE, the new default should address it AFAIK

**Comment by oceanwalker** [comment]
First, I try to reproduce the attack, add some blank and illegal characters to the URL, then through forward proxy transmit the request, but ATS didn't reject this request.

Second, I just use forward proxy ability, and didn't use reverse proxy, so I wonder if this is impacted by this CVE?

**Comment by bryancall** [comment]
@oceanwalker Can you please send an email to security@trafficserver.apache.org on how you are trying to reproduce this issue?

**Comment by oceanwalker** [comment]
![image](https://user-images.githubusercontent.com/43021640/178206173-a0669ea5-17a4-4e26-99ae-6d3b20b92dda.png)

![无标题](https://user-images.githubusercontent.com/43021640/178206556-b9495d66-032b-4c54-bafa-52b2c798e51e.png)

Based on the version 8.1.4 and used Xshell as SSH client, after adding invalid and blank characters shown in the above screenshot, the request is still  forwarded, which is contrary to our expectation.

**Comment by oceanwalker** [comment]
@bryancall  Considering it is so important to me and a lot of work would based on your answer, could you please share with me about the latest progress as early as you could. Thank you so much!

**Comment by bryancall** [comment]
The `proxy.config.http.strict_uri_parsing` setting applies to all incoming requests, including forward proxy. The default was changed to `2` in [PR #8632](https://github.com/apache/trafficserver/pull/8632), which rejects requests containing whitespace or non-printable characters in the URL. This has been verified with forward proxy testing on ATS master (b5557e690, March 6 2026) built with AddressSanitizer — no memory issues detected.

If you are still seeing issues with URL validation in forward proxy mode, please open a new issue with reproduction steps.
